org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/TextEditAcceptor.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2022 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server;
+
+/**
+ * @author Rub�n Porras Campo - Initial contribution and API, factored out of ChangeConverter2
+ *
+ * @since 2.27
+ */
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.VersionedTextDocumentIdentifier;
+import org.eclipse.lsp4j.WorkspaceClientCapabilities;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.lsp4j.WorkspaceEditCapabilities;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+import org.eclipse.xtext.ide.server.Document;
+import org.eclipse.xtext.ide.server.ILanguageServerAccess;
+
+public class TextEditAcceptor {
+	private final boolean useDocumentChanges;
+	private final WorkspaceEdit edit;
+
+	public TextEditAcceptor(final WorkspaceEdit edit, final ILanguageServerAccess access) {
```

Comment:
No need for final parameters

Reply:
done, we have different policy according to tabs and spaces and usages of final at Avaloq, and these slipped through when coding.